zebra-consensus/src/checkpoint/types.rs

```diff
@@ -62,6 +62,31 @@ impl PartialOrd for Progress<block::Height> {
     }
 }
 
+impl Progress<block::Height> {
+    /// Returns the contained height, or `None` if the progress has finished, or has not started.
+    pub fn height(&self) -> Option<block::Height> {
+        match self {
+            BeforeGenesis => None,
+            InitialTip(height) => Some(*height),
+            PreviousCheckpoint(height) => Some(*height),
+            FinalCheckpoint => None,
```

Comment:
Nitpick/Optional: Condense these match arms.

```suggestion
            BeforeGenesis | FinalCheckpoint => None,
            InitialTip(height) | PreviousCheckpoint(height) => Some(*height),
```

Reply:
I think it's easier to read as it is, because the variants are in chain order.